Compilation issue on >= 0.12.16 with `--no-default-features`

0.12.16 has introduced a compilation issue when `reqwest` is compiled with `--no-default-features`

```
Checking reqwest v0.12.17
error[E0432]: unresolved import `super::Scheme`
    --> /home/zerosnacks/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/reqwest-0.12.17/src/connect.rs:1221:27
     |
1221 |     use super::{BoxError, Scheme};
     |                           ^^^^^^ no `Scheme` in `connect`
     |
     = help: consider importing this struct instead:
             http::uri::Scheme
note: found an item that was configured out
    --> /home/zerosnacks/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/reqwest-0.12.17/src/connect.rs:4:16
     |
4    | use http::uri::Scheme;
     |                ^^^^^^
note: the item is gated behind the `__tls` feature
    --> /home/zerosnacks/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/reqwest-0.12.17/src/connect.rs:3:7
     |
3    | #[cfg(feature = "__tls")]
     |       ^^^^^^^^^^^^^^^^^
```

Reference: https://github.com/foundry-rs/compilers/actions/runs/15303809815/job/43051188178

Pinning to `=0.12.15` is confirmed to work

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -19,6 +19,7 @@ jobs:
     needs:
       - style
       - test
+      - features
       - unstable
       - nightly
       - msrv
@@ -200,6 +201,27 @@ jobs:
           cargo nextest run --locked --workspace ${{ matrix.features }} ${{ matrix.test-features }}
           cargo test --locked --workspace --doc ${{ matrix.features }} ${{ matrix.test-features }}
         shell: bash
+
+  features:
+    name: features
+    needs: [style]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Install cargo-hack
+        uses: taiki-e/install-action@cargo-hack
+
+      - uses: Swatinem/rust-cache@v2
+
+      - name: check --feature-powerset
+        run: cargo hack --no-dev-deps check --feature-powerset --depth 2 --skip http3,__tls,__rustls,__rustls-ring,native-tls-vendored,trust-dns
+        env:
+          RUSTFLAGS: "-D dead_code -D unused_imports"
 
   unstable:
     name: "unstable features"

--- a/src/connect.rs
+++ b/src/connect.rs
@@ -1214,11 +1214,12 @@ mod rustls_tls_conn {
 mod socks {
     use std::io;
 
+    use http::uri::Scheme;
     use http::Uri;
     use tokio::net::TcpStream;
     use tokio_socks::tcp::{Socks4Stream, Socks5Stream};
 
-    use super::{BoxError, Scheme};
+    use super::BoxError;
     use crate::proxy::Intercepted;
 
     pub(super) enum DnsResolve {

